src/Cores/Storage/src/FileMetadata/Scanners/DepthFirstFolderScanner.cs

```diff
@@ -82,42 +82,55 @@ await foreach (var subFile in RecursiveScanForFilesAsync(folder))
     private async Task EnableFolderWatcherAsync(IMutableFolder folder, CancellationToken cancellationToken)
     {
         cancellationToken.ThrowIfCancellationRequested();
-        var folderWatcher = await folder.GetFolderWatcherAsync(cancellationToken);
+        try
+        {
+            var folderWatcher = await folder.GetFolderWatcherAsync(cancellationToken);
 
-        cancellationToken.ThrowIfCancellationRequested();
+            cancellationToken.ThrowIfCancellationRequested();
 
-        lock (_knownSubFolders)
+            lock (_knownSubFolders)
+            {
+                if (folder.Id == RootFolder.Id)
+                    _rootFolderWatcher = folderWatcher;
+                else
+                    _knownSubFolders[folder.Id].FolderWatcher = folderWatcher;
+            }
+
+            folderWatcher.CollectionChanged += FolderWatcherOnCollectionChanged;
+        }
+        catch (System.Exception ex)
         {
-            if (folder.Id == RootFolder.Id)
-                _rootFolderWatcher = folderWatcher;
-            else
-                _knownSubFolders[folder.Id].FolderWatcher = folderWatcher;
+            OwlCore.Diagnostics.Logger.LogError("Couldn't get folder watcher", ex);
         }
-
-        folderWatcher.CollectionChanged += FolderWatcherOnCollectionChanged;
     }
 
     private async Task DisableFolderWatcherAsync(IMutableFolder folder, CancellationToken cancellationToken)
     {
         cancellationToken.ThrowIfCancellationRequested();
+        try
+        {
+            var folderWatcher = folder.Id == RootFolder.Id ? _rootFolderWatcher : _knownSubFolders[folder.Id].FolderWatcher;
+            Guard.IsNotNull(folderWatcher);
 
-        var folderWatcher = folder.Id == RootFolder.Id ? _rootFolderWatcher : _knownSubFolders[folder.Id].FolderWatcher;
-        Guard.IsNotNull(folderWatcher);
-
-        folderWatcher.CollectionChanged -= FolderWatcherOnCollectionChanged;
+            folderWatcher.CollectionChanged -= FolderWatcherOnCollectionChanged;
 
-        if (folder.Id == RootFolder.Id)
-            _rootFolderWatcher = null;
-        else
-        {
-            lock (_knownSubFolders)
+            if (folder.Id == RootFolder.Id)
+                _rootFolderWatcher = null;
+            else
             {
-                _knownSubFolders[folder.Id].FolderWatcher = null;
+                lock (_knownSubFolders)
+                {
+                    _knownSubFolders[folder.Id].FolderWatcher = null;
+                }
             }
-        }
 
-        await folderWatcher.DisposeAsync();
-        cancellationToken.ThrowIfCancellationRequested();
+            await folderWatcher.DisposeAsync();
```

Comment:
Maybe dispose the watcher before setting the corresponding FileWatcher to null on line No. 123